src/utils/actions.js

```diff
@@ -57,63 +57,63 @@ const isObjectEmpty = (obj) => {
     return Object.keys(obj).length === 0 && obj.constructor === Object ;
 }
 
-export const authErrorHandler = (err, res) => (dispatch, state) => {
-    let code = err.status;
-    let msg = '';
+export const authErrorHandler = (
+    err,
+    res,
+    notifyErrorHandler = showMessage
+) => (dispatch, state) => {
+
+    let msg = "";
+    const code = err.status;
+
+    const notifyError = (title, html, type, callback) => {
+        const errorPayload = { httpCode: code, title, html, type };
+        notifyErrorHandler(errorPayload, callback);
+    };
 
     dispatch(stopLoading());
 
     switch (code) {
-        case 403:
-            let error_message = {
-                title: 'ERROR',
-                html: T.translate("errors.user_not_authz"),
-                type: 'error'
-            };
-
-            dispatch(showMessage( error_message, initLogOut ));
-            break;
         case 401:
-            let currentLocation = getCurrentPathName();
-            let clearing_session_state = isClearingSessionState();
-
-            dispatch({
-                type: CLEAR_SESSION_STATE,
-                payload: {}
-            });
-
-            if(!clearing_session_state) {
-                setSessionClearingState(true);
-                console.log('authErrorHandler 401 - re login');
-                doLogin(currentLocation);
+            const initLogin = () => {
```

Comment:
declared clousure, adds stack frames and memory overload can we define it outside ?